### PR TITLE
fix: Update rbac settings for cluster autoscaler

### PR DESCRIPTION
**Pull request by nilo19**
Update rbac settings for cluster autoscaler.

### CODE REVIEW

File: parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml [hunk 1]
```diff
@@ -1,84 +1,89 @@
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   labels:
     k8s-addon: cluster-autoscaler.addons.k8s.io
     k8s-app: cluster-autoscaler
-    kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (feiskyer) [inline parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml]: please keep these two lines.

File: parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml [hunk 2]
```diff
@@ -1,84 +1,89 @@
...
 ---
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: cluster-autoscaler
   labels:
     k8s-addon: cluster-autoscaler.addons.k8s.io
     k8s-app: cluster-autoscaler
-    kubernetes.io/cluster-service: "true"
```
Comment (feiskyer) [inline parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml]: same here

**Comment by CecileRobertMichon** [comment]
@nilo19 can you please update the commit message to `fix: Update rbac settings for cluster autoscaler`? Otherwise the commit won't be picked up by our release changelog script.

git commit --amend
git push -f

**Comment by feiskyer** [comment]
@nilo19 Seems the commit message is still wrong?

**Comment by nilo19** [comment]
> @nilo19 Seems the commit message is still wrong?

Have been updated.

File: parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml (hunk 1 above)
Comment (feiskyer) [reply parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml]: keep these two lines?

File: parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml [hunk 3]
```diff
@@ -19,39 +18,48 @@ metadata:
...
-- apiGroups: ["policy"]
-  resources: ["poddisruptionbudgets"]
-  verbs: ["watch","list"]
-- apiGroups: ["apps"]
-  resources: ["statefulsets","replicasets"]
-  verbs: ["watch","list","get"]
-- apiGroups: ["storage.k8s.io"]
-  resources: ["storageclasses"]
-  verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
```
Comment (feiskyer) [inline parts/k8s/containeraddons/kubernetesmasteraddons-cluster-autoscaler-deployment.yaml]: keep the indents for this part?

**Comment by feiskyer** [comment]
@nilo19 PR needs rebase

**Comment by nilo19** [comment]
> @nilo19 PR needs rebase

Have rebased the code.

**Comment by feiskyer** [comment]
@CecileRobertMichon Could you help to take a look again?

**Comment by mboersma** [comment]
@nilo19 thanks again for fixing this!

CI failed on the code generation check. You need to run `make generate` and then commit the generated file to this PR.

**Comment by feiskyer** [comment]
> CI failed on the code generation check. You need to run make generate and then commit the generated file to this PR.

probably need to rebase first, and make generate

**Comment by nilo19** [comment]
> > CI failed on the code generation check. You need to run make generate and then commit the generated file to this PR.
> 
> probably need to rebase first, and make generate

Have performed `make generate`, rebased the code and amend the commit.

**Comment by nilo19** [comment]
> @nilo19 thanks again for fixing this!
> 
> CI failed on the [code generation](https://github.com/Azure/aks-engine/blob/master/docs/community/developer-guide.md#pull-requests-and-generated-code) check. You need to run `make generate` and then commit the generated file to this PR.

Thanks a lot. Have done as told.

**Comment by mboersma** [comment]
@nilo19 I updated this PR–hope that's ok. There were two incorrect lines in the ServiceAccount definition, and the copy of the manifest in the 1.16 directory needed the same updates applied. cc: @feiskyer.

**Pull request merged**